### PR TITLE
Add ssl support to the module and change the connection url format

### PR DESCRIPTION
**Pull request by lahirusamith**
## Purpose
$title

Fixes ballerina-platform/ballerina-standard-library#1741
Fixes ballerina-platform/ballerina-standard-library#1740
Fixes ballerina-platform/ballerina-standard-library#1231
Fixes ballerina-platform/ballerina-standard-library#1672

## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: ballerina/tests/01-connection-init-test.bal [hunk 1]
```diff
@@ -108,18 +110,144 @@ function testWithOptionsWithErroneousSSL() returns error? {
...
+        options = options
+    );
+    test:assertEquals(oracledbClient.close(), (), "Initializing with ssl param fail");
+}
+
+// with all params and options with SSL both key and cert
+@test:Config {
+    groups:["connection"]
+}
+function testWithOptionsWithSSLWitKey() returns error? {
```
Comment (kaneeldias) [inline ballerina/tests/01-connection-init-test.bal:229]:
```suggestion
function testWithOptionsWithSSLWithKey() returns error? {
```

**Review by kaneeldias — APPROVED** [review]

File: ballerina/tests/01-connection-init-test.bal [hunk 2]
```diff
@@ -108,18 +110,144 @@ function testWithOptionsWithErroneousSSL() returns error? {
...
+        user = USER,
+        password = PASSWORD,
+        port = SSLPORT,
+        database = DATABASE,
+        options = options
+    );
+    test:assertTrue(oracledbClient is error);
+    if oracledbClient is sql:ApplicationError {
+        test:assertTrue(oracledbClient.message().startsWith("Error in SQL connector configuration: Failed to initialize pool: " +
+        "IO Error: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException"));
```
Comment (daneshk) [inline ballerina/tests/01-connection-init-test.bal:163]: Shall we improve the error message without passing java exception

File: ballerina/tests/01-connection-init-test.bal [hunk 3]
```diff
@@ -108,18 +110,144 @@ function testWithOptionsWithErroneousSSL() returns error? {
...
+    } else {
+        test:assertFail("Error ApplicatonError expected");
+    }
+}
+
+// with all params and options with SSL with correct port with incorrect password
+@test:Config {
+    groups:["connection"]
+}
+function testWithOptionsWithErroneousSSLCorrctPortWrongPW() returns error? {
```
Comment (daneshk) [inline ballerina/tests/01-connection-init-test.bal:173]:
```suggestion
function testWithOptionsWithErroneousSSLCorrectPortWrongPW() returns error? {
```

File: ballerina/tests/01-connection-init-test.bal [hunk 4]
```diff
@@ -108,18 +110,144 @@ function testWithOptionsWithErroneousSSL() returns error? {
...
+        user = USER,
+        password = PASSWORD,
+        port = SSLPORT,
+        database = DATABASE,
+        options = options
+    );
+    test:assertTrue(oracledbClient is error);
+    if oracledbClient is sql:ApplicationError {
+        test:assertTrue(oracledbClient.message().endsWith("The Network Adapter could not establish the connection Caused by " +
+        ":Unable to initialize ssl context."));
```
Comment (daneshk) [inline ballerina/tests/01-connection-init-test.bal:194]: Can we improve the message further. give exact error

File: ballerina/tests/01-connection-init-test.bal [hunk 5]
```diff
@@ -108,18 +110,144 @@ function testWithOptionsWithErroneousSSL() returns error? {
...
+    } else {
+        test:assertFail("Error ApplicatonError expected");
+    }
+}
+
+// with all params and options with Erroneous SSL with correct port
+@test:Config {
+    groups:["connection"]
+}
+function testWithOptionsWithErroneousSSLCorrctPort() returns error? {
```
Comment (daneshk) [inline ballerina/tests/01-connection-init-test.bal:139]:
```suggestion
function testWithOptionsWithErroneousSSLCorrectPort() returns error? {
```

File: ballerina/tests/01-connection-init-test.bal (hunk 2 above)
Comment (lahirusamith) [reply ballerina/tests/01-connection-init-test.bal:163]: An issue was created to track and resolve printing java class exceptions
https://github.com/ballerina-platform/ballerina-standard-library/issues/1775

File: ballerina/tests/01-connection-init-test.bal (hunk 4 above)
Comment (lahirusamith) [reply ballerina/tests/01-connection-init-test.bal:194]: This error is originated in sql module and created an issue to track and update the error messages. https://github.com/ballerina-platform/ballerina-standard-library/issues/1775

**Review by daneshk — APPROVED** [review]
LGTM

**Pull request merged**